test/webhook-apicoverage/resourcetree/updateCoverage_test.go

```diff
@@ -29,42 +29,27 @@ func TestSimpleStructValue(t *testing.T) {
 	}
 }
 
-func TestPtrValueAllCovered(t *testing.T) {
-	tree := getTestTree(ptrTypeName, reflect.TypeOf(ptrType{}))
-	tree.UpdateCoverage(reflect.ValueOf(getPtrTypeValueAllCovered()))
-	if err := verifyPtrValueAllCovered(tree.Root); err != nil {
-		t.Fatal(err)
-	}
-}
-
-func TestPtrValueSomeCovered(t *testing.T) {
-	tree := getTestTree(ptrTypeName, reflect.TypeOf(ptrType{}))
-	tree.UpdateCoverage(reflect.ValueOf(getPtrTypeValueSomeCovered()))
-	if err := verifyPtrValueSomeCovered(tree.Root); err != nil {
-		t.Fatal(err)
-	}
-}
-
-func TestArrValueAllCovered(t *testing.T) {
-	tree := getTestTree(arrayTypeName, reflect.TypeOf(arrayType{}))
-	tree.UpdateCoverage(reflect.ValueOf(getArrValueAllCovered()))
-	if err := verifyArryValueAllCovered(tree.Root); err != nil {
-		t.Fatal(err)
-	}
-}
-
-func TestArrValueSomeCovered(t *testing.T) {
-	tree := getTestTree(arrayTypeName, reflect.TypeOf(arrayType{}))
-	tree.UpdateCoverage(reflect.ValueOf(getArrValueSomeCovered()))
-	if err := verifyArrValueSomeCovered(tree.Root); err != nil {
-		t.Fatal(err)
-	}
-}
+func TestUpdateCoverage(t *testing.T) {
+	datas := []struct {
+		TestName string
+		name     string
+		typeI    interface{}
+		value    interface{}
+		f        func(NodeInterface) error
+	}{{"TestPtrValueAllCovered", ptrTypeName, ptrType{}, getPtrTypeValueAllCovered(), verifyPtrValueAllCovered},
```

Comment:
Sorry, I didn't quite mean that compact 😅 

What we want is something like:
```
{{
    Foo: "bar",
    Baz: "blah",
}, {
    Foo: "matt",
    Bar: "chao",
}}
```

Reply:
lol, done